x-pack/plugins/observability_solution/infra/server/routes/entities/get_data_stream_types.ts

```diff
@@ -56,7 +56,14 @@ export async function getDataStreamTypes({
 
   if (latestEntity) {
     castArray(latestEntity.sourceDataStreamType).forEach((item) => {
-      sourceDataStreams.add(item as EntityDataStreamType);
+      if (
+        item !== null &&
+        [EntityDataStreamType.LOGS, EntityDataStreamType.METRICS].includes(
+          item as EntityDataStreamType
+        )
```

Comment:
Nit: This is probably enough but it can be simplified and we can do something like: 
```suggestion
        item in [EntityDataStreamType.LOGS, EntityDataStreamType.METRICS] 
```

```